tools/generator/dfg/stm32/stm.py

```diff
@@ -89,6 +89,71 @@ def getGpioRemapForModuleConfig(module, config):
         mmm['mapping'] = stm32f1_gpio_remap[module]['mapping'][int(config)]
     return mmm
 
+stm32f3_dma_remap = \
+{
+    'dma1ch1': {
+        'tim17_ch1': 'tim17_up',
+        'tim17_up':  {'position': 12, 'clear': 1, 'set': 0},
+    },
+    'dma1ch2': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 2},
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 1},
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 0}, # also 'set': 3
+    },
+    'dma1ch3': {
+        'dac1_ch1':  'tim6_up',
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 1},
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 0}, # also 'set': 3
+        'tim16_ch1': 'tim16_up',
+        'tim16_up':  {'position': 11, 'clear': 1, 'set': 0},
+        'tim6_up':   {'position': 13, 'clear': 1, 'set': 1},
+    },
+    'dma1ch4': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 3},
```

Comment:
Oh gawd, this is a gem, see dma2ch1. Note that this writes both bit 72 and bit 73.